src/plugins/dashboard/public/application/actions/unlink_from_library_action.tsx

```diff
@@ -0,0 +1,92 @@
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { i18n } from '@kbn/i18n';
+import _ from 'lodash';
+import uuid from 'uuid';
+import { ActionByType, IncompatibleActionError } from '../../ui_actions_plugin';
+import { ViewMode, PanelState, IEmbeddable } from '../../embeddable_plugin';
+import {
+  PanelNotFoundError,
+  EmbeddableInput,
+  isReferenceOrValueEmbeddable,
+} from '../../../../embeddable/public';
+import { DashboardPanelState, DASHBOARD_CONTAINER_TYPE, DashboardContainer } from '..';
+
+export const ACTION_UNLINK_FROM_LIBRARY = 'unlinkFromLibrary';
+
+export interface UnlinkFromLibraryActionContext {
+  embeddable: IEmbeddable;
+}
+
+export class UnlinkFromLibraryAction implements ActionByType<typeof ACTION_UNLINK_FROM_LIBRARY> {
+  public readonly type = ACTION_UNLINK_FROM_LIBRARY;
+  public readonly id = ACTION_UNLINK_FROM_LIBRARY;
+  public order = 15;
+
+  constructor() {}
+
+  public getDisplayName({ embeddable }: UnlinkFromLibraryActionContext) {
+    if (!embeddable.getRoot() || !embeddable.getRoot().isContainer) {
+      throw new IncompatibleActionError();
+    }
+    return i18n.translate('dashboard.panel.unlinkFromLibrary', {
+      defaultMessage: 'Unlink from library item',
+    });
+  }
+
+  public getIconType({ embeddable }: UnlinkFromLibraryActionContext) {
+    if (!embeddable.getRoot() || !embeddable.getRoot().isContainer) {
+      throw new IncompatibleActionError();
+    }
+    return 'folderExclamation';
+  }
+
+  public async isCompatible({ embeddable }: UnlinkFromLibraryActionContext) {
+    return Boolean(
+      embeddable.getInput()?.viewMode !== ViewMode.VIEW &&
+        embeddable.getRoot() &&
+        embeddable.getRoot().isContainer &&
+        embeddable.getRoot().type === DASHBOARD_CONTAINER_TYPE &&
+        isReferenceOrValueEmbeddable(embeddable) &&
+        embeddable.inputIsRefType(embeddable.getInput())
+    );
+  }
+
+  public async execute({ embeddable }: UnlinkFromLibraryActionContext) {
+    if (!isReferenceOrValueEmbeddable(embeddable)) {
+      throw new IncompatibleActionError();
+    }
+
+    const newInput = await embeddable.getInputAsValueType();
+    embeddable.updateInput(newInput);
+
+    const dashboard = embeddable.getRoot() as DashboardContainer;
+    const panelToReplace = dashboard.getInput().panels[embeddable.id] as DashboardPanelState;
+    if (!panelToReplace) {
+      throw new PanelNotFoundError();
+    }
+
+    const newPanel: PanelState<EmbeddableInput> = {
+      type: embeddable.type,
+      explicitInput: { ...newInput, id: uuid.v4() },
+    };
+    dashboard.replacePanel(panelToReplace, newPanel);
```

Comment:
not related to this PR, but really shows how much we need an `updatePanel` rather than `replacePanel` functionality 😞

Reply:
I absolutely agree, fingers crossed that this https://github.com/elastic/kibana/pull/74253 will handle it.